pkg/repo/mysql/translation.go

```diff
@@ -14,22 +14,25 @@ import (
 )
 
 func (r *Repo) SaveTranslation(ctx context.Context, serviceID uuid.UUID, translation *model.Translation) error {
-	_, err := r.LoadService(ctx, serviceID)
-	if err != nil {
-		return fmt.Errorf("repo: load service: %w", err)
+	if _, ok := r.db.(*sql.Tx); ok { // use existing tx
+		return r.saveTranslation(ctx, serviceID, translation)
 	}
 
-	tx, err := r.db.BeginTx(ctx, nil)
+	return r.Tx(ctx, func(ctx context.Context, rp repo.Repo) error { // create new tx
+		return rp.SaveTranslation(ctx, serviceID, translation) //nolint:wrapcheck
+	})
+}
+
+func (r *Repo) saveTranslation(ctx context.Context, serviceID uuid.UUID, translation *model.Translation) error {
```

Comment:
The same here, `saveTranslation` is not required, called only once in `SaveTranslation`.